### PR TITLE
New blog post - TIL Jekyll and GitHub Pages

### PR DESCRIPTION
**Pull request by ThatRobVK**
Added a new blog post about learning Jekyll and GitHub Pages to host a blog

### CODE REVIEW

File: _posts/2019-04-15-TIL-Jekyll-and-GitHub-Pages.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+Meanwhile, I notice a sentence `before configuring your custom domain with your DNS provider, add your custom domain to your GitHub Pages site` - Whoops! Let's quickly do that. There's a [GitHub quick start guide](https://help.github.com/en/articles/quick-start-setting-up-a-custom-domain) that shows you how to do it.
+
+![GitHub custom domain configuration](/images/2019-04-15-CustomDomain.png)
+
+You can only add a single domain through the UI. So what about `www` and my blog subdomain? Let's see what it's done to the files. When I do a `git pull` I get a `CNAME` file down with `robvk.uk` as its contents.
+
+At this point I spent a little time digging into whether I could get the DNS for the subdomain to point to a subfolder on the primary domain. I'm pretty sure DNS won't do this and normally you'd configure the web server to route traffic by domain to a website on the server. I did read an amusing thread on a forum with one person becoming quite irate whilst repeatedly claiming their work DNS did exactly this. I gave it a go anyway, just in case. Even though Route53 accepts a full URL as the target for a `CNAME`, it doesn't work.
+
+{: .um-actually}
```
Comment (florx) [inline _posts/2019-04-15-TIL-Jekyll-and-GitHub-Pages.md]: does this work on the IW blog?

File: _posts/2019-04-15-TIL-Jekyll-and-GitHub-Pages.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+But I now have an idea. I create another repository called `TodayILearned`. In the repository settings I go down to GitHub Pages and switch it to serve from the `master` branch. It doesn't allow you to select a domain name here, but if you upload the `CNAME` file we created earlier for `til.robvk.uk` it should work. I leave my DNS setup as-is as it already had the subdomain pointing at GitHub.
+
+![Settings of the Today I Learned repo](/images/2019-04-15-TilRepoSettings.png)
+
+So why an `A` for the root and `CNAME` for TIL? According to [the GitHub docs](https://help.github.com/en/articles/setting-up-an-apex-domain#configuring-an-alias-or-aname-record-with-your-dns-provider) you can use a CNAME for the root but this can cause problems with things like e-mail. For this to work correctly, your DNS provider needs to support something called "CNAME Flattening". I don't have time to dig into what that is and whether Route53 supports it right now, so using an `A` record is the easiest way forward.
+
+{: .box-note}
```
Comment (florx) [inline _posts/2019-04-15-TIL-Jekyll-and-GitHub-Pages.md]: also this seems off

File: _posts/2019-04-15-TIL-Jekyll-and-GitHub-Pages.md (hunk 1 above)
Comment (ThatRobVK) [reply _posts/2019-04-15-TIL-Jekyll-and-GitHub-Pages.md]: Do you mean the {: .um-actually}? No it doesn't, but ends up putting in a class that doesn't exist. I'll update it to remove that line, it's cleaner.

Comment (ThatRobVK) [reply _posts/2019-04-15-TIL-Jekyll-and-GitHub-Pages.md]: I have removed the tag

File: _posts/2019-04-15-TIL-Jekyll-and-GitHub-Pages.md (hunk 2 above)
Comment (ThatRobVK) [reply _posts/2019-04-15-TIL-Jekyll-and-GitHub-Pages.md]: Removed the tag

**Pull request merged**